### ISSUE TITLE
Output Window doesn't always scroll to bottom when new output is added

### ISSUE THREAD

**Issue by dannyparsons** [africanmathsinitiative/R-Instat#6996]
If you click somewhere at the top of the output window so that the cursor is within one of the output blocks.
Then go to a dialog that produces output. It doesn't scroll to the end of the window. I think it just scrolls to the end of the block that it is in.
So it works if the cursor wasn't in the output window before running the dialog. But not if you were somewhere within the output.

**Comment by ChrisMarsh82** [comment]
The issue is due to do with the cursor staying in a richTextBox. I have tried to change focus and the activeControl but neither has worked. The only other possible workaround I have found is to tab through the controls until the last textbox is selected, using SendKeys.Send("{TAB}"). This is not a great solution and may cause issues as it is sending [TAB] to the PC and not limited to the program.

**Comment by dannyparsons** [comment]
@Patowhiz Could you investigate possible solutions to this? Don't spend too long, but a few hours would be worthwhile. If there's no better solution then I suggest implementing the Tab method Chris suggests.

**Comment by dannyparsons** [comment]
@Patowhiz This is a much more urgent issue than most others at the moment. Can you please prioritise this one?

**Comment by Patowhiz** [comment]
@dannyparsons yes I'm on it.

**Comment by dannyparsons** [comment]
@Patowhiz Please keep this at the top of your R-Instat to do list.

**Comment by Patowhiz** [comment]
@dannyparsons @ChrisMarsh82 I have been looking at this `annoying` issue and trying to understand the why.
@ChrisMarsh82 the reason as to why changing the `Focus` doesn't work it's because of the `ActiveControl`. When Ok is clicked, the `ActiveControl` in the `frmMain` is the `SplitContainer`. My follow up shows that calling `Focus` on a control that is not `ActiveControl` fails (returns false). 

However, I noticed that, when I set the `ActiveControl` to `Nothing` before calling the `anyDialog.ShowDialog()`, changing `Focus` in the output window works.

@dannyparsons based on what I have so far we have 2 options;
1. Either go with `SendKeys.Send("{TAB}")`  or
2. Set the `ActiveControl` to `Nothing` before showing any dialog and changing the `Focus` to the panel of the last output.

@ChrisMarsh82 going by the `TAB` option, my understanding is, I will have to `SendKeys.Send("{TAB}")` from the point of focus all through the last output? And I may have to be careful that this is done in the correct event/time, probably after the dialog is hidden (Currently, the output window receives some outputs(the R script) before the dialog is fully closed).

**Comment by Patowhiz** [comment]
@dannyparsons @ChrisMarsh82 option 2 sample demonstration code is in draft PR #7219.
I can also do a draft PR for option 1, if that's desired.

**Comment by Patowhiz** [comment]
@dannyparsons yes I think it can be generalised. 

> we don't need to do this in every place a dialog is shown

That bit is what I have to think about.
Have a look at PR #7219.  We only need this done at `frmMain` level, not in any other form

**Comment by dannyparsons** [comment]
Yes, but even if it's just in this file, if it has to be done everywhere a dialog is opened I'm not so keen, that's prone to mistakes and a lot of repeated code.

Is it possible to do this, maybe in ucrBase, either before the parent form loads or after it closes? Might not be possible from within a control, but that's the sort of place we want this kind of thing to be.

**Comment by Patowhiz** [comment]
@dannyparsons I agree.

I couldn't find a way to generalise option 2.  The only way was to wrap any `showDialog()` call. And that meant going through all lines calling that function. Big change and probably not your preference. 

Doing option 1 proved too hacky, I had to loop through all the main panel controls up to the last one while calling `SendKeys.Send("{TAB}"`, and I could still see its platform level effects when R-Instat wasn't the active application. And where to make that call just seemed to hacky and not viable.

So I relooked at the original problem, which is 

> the reason as to why changing the `Focus `doesn't work it's because of the `ActiveControl.` When Ok is clicked, the `ActiveControl `in the `frmMain `is the `SplitContainer`. My follow up shows that calling Focus on a control that is not `ActiveControl` fails (returns false).

Digging deeper, I then realised that VB.Net events seem not to be bubbling all the way to parent level. 
So the events change have to be localised at the child level.  Which gave me option 3;

Return the focus from the any of the main panel children back to the main panel, and this needs to be done before the `ActiveControl` control change. Which effectively returns `ActiveControl` back to the main panel of the `outputpage`, thus giving it back control of all possible active events, , for instance `scroll` event.

In short PR #7219 now fixes this in a way that I think is not too hacky and is sufficient for what we need.

Thanks
